pilot/pkg/model/destination_rule.go

```diff
@@ -56,10 +61,15 @@ func (ps *PushContext) mergeDestinationRule(p *processedDestRules, destRuleConfi
 		if mergedRule.TrafficPolicy == nil && rule.TrafficPolicy != nil {
 			mergedRule.TrafficPolicy = rule.TrafficPolicy
 		}
+		if len(p.exportTo[resolvedHost]) == 0 && len(exportToMap) > 0 {
+			// we have dest rule in same namespace for the same host with updated exportTo. use it.
+			p.exportTo[resolvedHost] = exportToMap
```

Comment:
It is. But to support this use case, we will need to have two level data structure. One by the resolvedHost and then another by each exportTo namespace. My worry is that it may be a bit too complicated for user to grok. Easier to simply state that they should have only one dest rule for a host in a namespace and that if they have many, we will merge them.

Reply:
But the comments is not quite right.

Reply:
You are right. I fixed the comments in this file..